lib/model.js

```diff
@@ -333,6 +333,13 @@ ModelBaseClass.prototype._initProperties = function(data, options) {
     if (propVal === undefined && persistUndefinedAsNull) {
       self.__data[p] = propVal = null;
     }
+    // coerce array values of primitives from string
+    if (Array.isArray(type) && type[0].name) {
```

Comment:
What if a non-primitive type is nested deeper? I am thinking about something like this:

```js
type = [
  // an array of anonymous objects with the following structure:
  {
    // name is a string
    name: String,
    // roles hold an array of Role instances
    roles: [Role]
  }
]
```

Do we need to handle this case?

If anonymous objects are converted into named models, then how about array of arrays?

```js
// Cell is a model, let's say it has one property "color"

// 2D array of Cell instances
type = [[Cell]]; 

// Example value:
propValue = [
 [{color: 'black'}, {color: 'white'}],
 [{color: 'white'}, {color: 'black'}],
];
```

Reply:
I would like to tackle this in a follow-up PR. Will create an issue for it.

Reply:
@b-admike have you created that issue yet? Please post a link to it here.

Reply:
Yes I've created it now :) https://github.com/strongloop/loopback-datasource-juggler/issues/1716.

Reply:
I think this case should be fixed now since the actual bug was the fact that we didn't coerce array types of dates properly, but we do traverse through nested properties properly. As part of #1716, we should probably write tests first to verify before working on a fix.